Addons/tsgExoticQuadrature.hpp

```diff
@@ -168,9 +181,28 @@ inline TasGrid::CustomTabulated getExoticQuadrature(const int n, const double sh
 }
 
 // Basic function overloads.
-inline TasGrid::CustomTabulated getExoticQuadrature(const int n, const double shift,
-                                                    std::function<double(double)> weight_fn) {
-    return getExoticQuadrature(n, shift, weight_fn, 50*n+1, "Exotic quadrature");
+
+// If only the weight function is provided, the reference points and weights are generated by the GL rule and the weight_fn values
+// are generated by applying the weight function to these reference points.
+inline TasGrid::CustomTabulated getExoticQuadrature(const int n, const double shift, std::function<double(double)> weight_fn,
+                                                    const int nref, const char* description, const bool is_symmetric = false) {
+    std::vector<double> shifted_weights, ref_points, weight_fn_vals(nref);
+    TasGrid::OneDimensionalNodes::getGaussLegendre(nref, shifted_weights, ref_points);
+    std::transform(ref_points.begin(), ref_points.end(), weight_fn_vals.begin(), [weight_fn](double x){return weight_fn(x);});
```

Comment:
Do you need to wrap a lambda inside a lambda? Can you just pass in `weight_fn()` directly?

Reply:
Good point. I changed it to pass ``weight_fn`` in directly.